src/screens/DiscoveryScreenV2/index.js

```diff
@@ -61,63 +67,120 @@ const DiscoveryScreenV2 = ({route}) => {
+        setDiscoveryDataUnfollowedUsers(
+          data?.unfollowedUsers?.map((item) => ({
+            ...item,
+            following: item.user_id_follower !== null
+          })) || []
+        );
       }
       setIsLoadingDiscovery((prevState) => ({
         ...prevState,
         user: false
       }));
-    });
+    };
+
+    const fetchDiscoveryInitialDomains = async () => {
+      const initialData = await DiscoveryRepo.fetchInitialDiscoveryDomains();
+      FollowingAction.setFollowingDomain(initialData.suggestedDomains, followingDispatch);
+      DiscoveryAction.setDiscoveryInitialDomains(initialData.suggestedDomains, discoveryDispatch);
+    };
 
-    DiscoveryRepo.fetchDiscoveryDataDomain(text, {cancelToken}).then(async (data) => {
+    const fetchDiscoveryDataDomain = async () => {
+      const cancelToken = cancelTokenRef?.current?.token;
+      const data = await DiscoveryRepo.fetchDiscoveryDataDomain(text, {cancelToken});
       if (data.success) {
         setDiscoveryDataFollowedDomains(data?.followedDomains || []);
         setDiscoveryDataUnfollowedDomains(data?.unfollowedDomains || []);
       }
-
       setIsLoadingDiscovery((prevState) => ({
         ...prevState,
         domain: false
       }));
-    });
+    };
+
+    const fetchDiscoveryInitialTopics = async () => {
+      const initialData = await DiscoveryRepo.fetchInitialDiscoveryTopics();
+      DiscoveryAction.setDiscoveryInitialTopics(initialData.suggestedTopics, discoveryDispatch);
+    };
 
-    DiscoveryRepo.fetchDiscoveryDataTopic(text, {cancelToken}).then(async (data) => {
+    const fetchDiscoveryDataTopic = async () => {
+      const cancelToken = cancelTokenRef?.current?.token;
+      const data = await DiscoveryRepo.fetchDiscoveryDataTopic(text, {cancelToken});
       if (data.success) {
         setDiscoveryDataFollowedTopics(data?.followedTopic);
         setDiscoveryDataUnfollowedTopics(data?.unfollowedTopic);
       }
-
       setIsLoadingDiscovery((prevState) => ({
         ...prevState,
         topic: false
       }));
-    });
+    };
 
-    DiscoveryRepo.fetchDiscoveryDataNews(text, {cancelToken}).then(async (data) => {
+    const fetchDiscoveryDataNews = async () => {
+      const cancelToken = cancelTokenRef?.current?.token;
+      const data = await DiscoveryRepo.fetchDiscoveryDataNews(text, {cancelToken});
       if (data.success) {
         setDiscoveryDataNews(data?.news);
       }
-
       setIsLoadingDiscovery((prevState) => ({
         ...prevState,
         news: false
       }));
-    });
+    };
+
+    if (text === undefined) {
+      await Promise.all([
+        fetchDiscoveryInitialUsers(),
+        fetchDiscoveryInitialDomains(),
+        fetchDiscoveryInitialTopics(),
+        fetchDiscoveryDataNews()
+      ]);
+    } else {
+      await Promise.all([
+        fetchDiscoveryDataUser(),
+        fetchDiscoveryDataDomain(),
+        fetchDiscoveryDataTopic(),
+        fetchDiscoveryDataNews()
+      ]);
+    }
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `fetchDiscoveryData` function has been significantly refactored. It now includes several nested async functions for fetching different types of discovery data. The use of `Promise.all` to run these fetch operations concurrently is a good practice for performance. However, error handling seems to be missing in these async functions. Consider adding try-catch blocks to handle any potential errors during API calls.



<!-- This is an auto-generated comment by OSS CodeRabbit -->